pkg/sbom/scanner/scanner_test.go

```diff
@@ -240,6 +240,68 @@ func TestRetryLogic_Host(t *testing.T) {
 	case <-time.After(4 * time.Second):
 	}
 	cancel()
-	// Ensure the collector is stopped
+	shutdown.WaitUntil(time.After(5 * time.Second))
+}
+
+// Test retry handling in case of an error when sending the result to a full channel
+func TestRetryChannelFull(t *testing.T) {
+	// Create a workload meta global store
+	workloadmetaStore := fxutil.Test[workloadmeta.Mock](t, fx.Options(
+		logimpl.MockModule(),
+		compConfig.MockModule(),
+		fx.Supply(context.Background()),
+		fx.Supply(workloadmeta.NewParams()),
+		workloadmeta.MockModuleV2(),
+	))
+
+	// Store the image
+	imageID := "id"
+	workloadmetaStore.Set(&workloadmeta.ContainerImageMetadata{
+		EntityID: workloadmeta.EntityID{
+			ID:   imageID,
+			Kind: workloadmeta.KindContainerImageMetadata,
+		},
+	})
+
+	// Create a mock collector
+	collName := "mock"
+	mockCollector := collectors.NewMockCollector()
+	resultCh := make(chan sbom.ScanResult)
+	expectedResult := sbom.ScanResult{Report: mockReport{id: imageID}}
+	mockCollector.On("Options").Return(sbom.ScanOptions{})
+	mockCollector.On("Scan", mock.Anything, mock.Anything).Return(expectedResult)
+	mockCollector.On("Channel").Return(resultCh)
+	shutdown := mockCollector.On("Shutdown")
+	mockCollector.On("Type").Return(collectors.ContainerImageScanType)
+
+	// Set up the configuration
+	cfg := config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	cfg.Set("sbom.scan_queue.base_backoff", "2s", model.SourceAgentRuntime)
+	cfg.Set("sbom.scan_queue.max_backoff", "5s", model.SourceAgentRuntime)
+
+	// Create a scanner and start it
+	scanner := NewScanner(cfg, map[string]collectors.Collector{collName: mockCollector}, optional.NewOption[workloadmeta.Component](workloadmetaStore))
+	ctx, cancel := context.WithCancel(context.Background())
+	scanner.Start(ctx)
+
+	// Enqueue a scan request for container images
+	err := scanner.Scan(sbom.ScanRequest(&scanRequest{collectorName: collName, id: imageID, scanRequestType: sbom.ScanFilesystemType}))
+	assert.NoError(t, err)
+
+	// Wait long enough for the `sendResult` function to fail
+	time.Sleep(sendTimeout + 1*time.Second)
+
+	// Make sure we recover
+	res := <-resultCh
+	assert.Equal(t, expectedResult.Report, res.Report)
+
+	// Make sure we don't receive anything afterward
+	select {
+	case res := <-resultCh:
+		t.Errorf("unexpected result received %v", res)
+	case <-time.After(4 * time.Second):
```

Comment:
The problem that I see with this test is that it's going to wait 4 seconds here plus 6 in the Sleep above. I don't have enough context about this code to propose an alternative, so I'll let you decide what to do about it.

Reply:
I reduced it to a few ms for every tests. I think it's better.